src/features/PostState/usePosts.js

```diff
@@ -0,0 +1,82 @@
+import { createContext, useContext, useEffect, useReducer } from 'react';
+import postsReducer from './postsReducer';
+import Airtable from 'airtable';
+
+export const PostsContext = createContext();
```

Comment:
The general pattern for using a context provider:

```js
// usePosts.js
import { createContext, useContext } from 'react';

export const PostsContext = createContext();

export const PostsProvider = ({ children }) => {
    // define all the custom business and state management logic here
   const stateAndFunctionsYouWantToExport = {}
    return (
        <PostsContext.Provider
            value={stateAndFunctionsYouWantToExport}
        >
            {children}
        </PostsContext.Provider>
    );
};

const usePosts = () => useContext(PostsContext);

export default usePosts;
```

In the component file highest in the component tree you want to use the context

```js

const PostsWithContext = () => (
    <PostsProvider>
        <Posts />
    </PostsProvider>
);
```

Then `<Posts/>` and any of its children have access to the `<PostsProvider>` context! In the `Posts.js` file:

```js
import usePosts from 'features/PostState/usePosts';

const Posts = () => {
    const stateAndFunctionsYouWantToExport = usePosts();
    // use the context!
    return
}
```